### PR TITLE
Nazar/feture/refactoring

### PR DESCRIPTION
**Pull request by Nazar-kob**

### CODE REVIEW

**Review by StoliarskyiKostiantyn — APPROVED** [review]

File: .vscode/settings.json [hunk 1]
```diff
@@ -27,6 +27,6 @@
   "python.testing.unittestEnabled": false,
   "python.testing.pytestArgs": ["tests"],
   "[python]": {
-    "editor.defaultFormatter": "ms-python.black-formatter"
+    "editor.defaultFormatter": "ms-python.python"
```
Comment (nixissmtp) [inline .vscode/settings.json]: return black formatter

File: app/views/case.py [hunk 2]
```diff
@@ -55,7 +56,7 @@ def get_all():
     )
 
 
-@bp.route("/get/<int:id>", methods=["GET"])
+@bp.route("/<int:id>", methods=["GET"])
 @login_required
 def get(id: int):
```
Comment (nixissmtp) [inline app/views/case.py]: replace get by get_case

File: app/views/case.py [hunk 3]
```diff
@@ -172,26 +168,26 @@ def create():
     return redirect(url_for("case.get_all"))
 
 
-@bp.route("/update", methods=["POST"])
+@bp.route("/update-status/<int:id>", methods=["PATCH"])
 @login_required
-def update():
+def update(id: int):
```
Comment (nixissmtp) [inline app/views/case.py]: update -> update_case

File: app/views/case.py [hunk 4]
```diff
@@ -106,58 +107,53 @@ def create():
...
+        session.add(new_case)
+        session.commit()
+        session.refresh(new_case)
+        ActionLogs.create_case_log(m.ActionsType.CREATE, new_case.id)
+
+        for index, img in enumerate(screenshots):
+            new_screenshot = m.CaseScreenshot(
+                url=img,
+                case_id=new_case.id,
+                origin_file_name=f"Screenshot {index}",
```
Comment (nixissmtp) [inline app/views/case.py]: f"Screenshot {index}" -> f"screenshot_{index}"

File: app/views/case.py [hunk 5]
```diff
@@ -204,7 +200,7 @@ def update():
 @login_required
 def delete(id: int):
     case = db.session.get(m.Case, id)
-    if not case:
+    if not case or case.is_deleted:
         log(log.INFO, "There is no case with id: [%s]", id)
         flash("There is no such case", "danger")
```
Comment (nixissmtp) [inline app/views/case.py:205]: remove flash from ajax request

File: src/admin.ts [hunk 6]
```diff
@@ -38,14 +47,48 @@ export const admin = () => {
...
-          let id = button.getAttribute('data-admin-id');
-          const response = await fetch(`/admin/delete/${id}`, {
-            method: 'DELETE',
-          });
-          if (response.status == 200) {
-            location.reload();
-          }
+        console.log('delete admin');
+        confirmModal.show();
+        const id = button.getAttribute('data-admin-id');
```
Comment (nixissmtp) [inline src/admin.ts:51]: add parseInt

Comment (nixissmtp) [reply src/admin.ts:51]: const id: number =

File: src/candidate.ts [hunk 7]
```diff
@@ -6,26 +17,64 @@ export const candidate = () => {
...
   if (searchInputButton && searchInput) {
     searchInputButton.addEventListener('click', () => {
       const url = new URL(window.location.href);
       url.searchParams.set('q', searchInput.value);
       window.location.href = `${url.href}`;
     });
   }
-  // delte flow
+  // delete candidate
   const deleteButtons = document.querySelectorAll('#delete-candidate-btn');
```
Comment (nixissmtp) [inline src/candidate.ts:35]: remove if statement

File: src/case.ts [hunk 8]
```diff
@@ -27,13 +30,98 @@ export const cases = () => {
...
+
+  const confirmCaseModal: ModalInterface = new Modal(
+    $confirmCaseModalElement,
+    modalOptions,
+  );
+
+  // callBack on btn is_active and is_main
+  const caseConfirmModalListener = (
+    event: MouseEvent,
+    btnElement: HTMLInputElement,
```
Comment (nixissmtp) [inline src/case.ts]: btnElement -> inputElement: HTMLInputElement

File: src/case.ts [hunk 9]
```diff
@@ -107,24 +220,52 @@ export const cases = () => {
 
       console.log(caseId);
 
-      const response = await fetch(`/case/get/${caseId}`, {
+      const response = await fetch(`/case/${caseId}`, {
         method: 'GET',
       });
       const caseData = await response.json();
```
Comment (nixissmtp) [inline src/case.ts:226]: add response interface/type

File: tests/test_case.py [hunk 10]
```diff
@@ -41,16 +41,22 @@ def test_crud_case(client, mocker):
...
-    # case: m.Case = db.session.get(m.Case, 1)
-    # assert case.is_active == test_case["is_active"]
-    # res = client.patch(f"/case/update/{case.id}", data={"field": "is_active"})
-    # assert not case.is_active
+    # update active and main
+    case: m.Case = db.session.get(m.Case, 1)
+    assert case.is_active == test_case["is_active"]
+    res = client.patch(f"/case/update-status/{case.id}", data={"field": "is_active"})
+    assert res.status_code == 200
+    action_log: m.Action = db.session.get(m.Action, 2)
```
Comment (nixissmtp) [inline tests/test_case.py]: remove hardcoded id

**Review by nixissmtp — CHANGES_REQUESTED** [review]

**Pull request merged**